rust-analyzer-unwrapped: 2024-09-02 -> 2024-11-11

khaneliman opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done
https://github.com/rust-lang/rust-analyzer/releases/tag/2024-11-04
https://github.com/rust-lang/rust-analyzer/releases/tag/2024-11-11

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

khaneliman committed:
rust-analyzer-unwrapped: 2024-09-02 -> 2024-11-04

khaneliman commented:
Need to look into the neovim test failure.

oxalica commented:
The r-a binary works for my nvim/coc.nvim setup, and I couldn't reproduce the failure by running the test script with `nix develop --ignore-environment` env. I guess it might be some other env issue which prevents r-a from finding the std src.

Could we disable that test for now?

khaneliman committed:
rust-analyzer/wrapper: format

khaneliman committed:
rust-analyzer/test-neovim-lsp: format

khaneliman committed:
rust-analyzer-unwrapped: format

khaneliman committed:
rust-analyzer-unwrapped: disable broken test

Currently fails to find std lib in neovim environment so hover doesn't
contain type information needed to successfully pass the test.

khaneliman commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 354304`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>ra-multiplex</li>
    <li>rust-analyzer</li>
    <li>rust-analyzer-unwrapped</li>
    <li>vscode-extensions.rust-lang.rust-analyzer</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>ra-multiplex</li>
    <li>rust-analyzer</li>
    <li>rust-analyzer-unwrapped</li>
    <li>vscode-extensions.rust-lang.rust-analyzer</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>ra-multiplex</li>
    <li>rust-analyzer</li>
    <li>rust-analyzer-unwrapped</li>
    <li>vscode-extensions.rust-lang.rust-analyzer</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>ra-multiplex</li>
    <li>rust-analyzer</li>
    <li>rust-analyzer-unwrapped</li>
    <li>vscode-extensions.rust-lang.rust-analyzer</li>
  </ul>
</details>

khaneliman commented:
> The r-a binary works for my nvim/coc.nvim setup, and I couldn't reproduce the failure by running the test script with `nix develop --ignore-environment` env. I guess it might be some other env issue which prevents r-a from finding the std src.
> 
> Could we disable that test for now?

Yeah, looks like something recently affected the test environment. I can recreate the test failing on master, too. I'm just disabling it, for now.

oxalica commented on pkgs/development/tools/rust/rust-analyzer/default.nix:
```diff
@@ -1,30 +1,41 @@
   pname = "rust-analyzer-unwrapped";
-  version = "2024-09-02";
-  cargoHash = "sha256-t45RzYkuywGByGWwUON3dW0aKjLYcFXB8uy4CybPuf4=";
+  version = "2024-11-04";
```
2024-11-11 is out now.

oxalica commented on pkgs/development/tools/rust/rust-analyzer/default.nix:
```diff
@@ -58,13 +69,16 @@ rustPlatform.buildRustPackage rec {
     updateScript = nix-update-script { };
     # FIXME: Pass overrided `rust-analyzer` once `buildRustPackage` also implements #119942
-    tests.neovim-lsp = callPackage ./test-neovim-lsp.nix { };
+    # tests.neovim-lsp = callPackage ./test-neovim-lsp.nix { };
```
Please add a `FIXME` comment with a link to this issue to explain why it is disabled.

oxalica commented on pkgs/development/tools/rust/rust-analyzer/test-neovim-lsp.nix:
```diff
@@ -1,62 +1,77 @@
-      var.push(Some("hello".to_owned()));
-    }
-  '';
+    testRustSrc = # Rust
+      ''
```
This inline comment messes up with formatting. Please remove it and nixfmt should format them with one less indentation.

khaneliman replied:
These are used for syntax highlighting with treesitter. It's your derivation, so I can... but, I find it helpful for looking at these scripts that are embedded in nix files.

oxalica replied:
> but, I find it helpful for looking at these scripts that are embedded in nix files.

I know that, but comments should not be inserted in an odd location where human will never do. I don't like optimizing machine readability with the cost of human readability. Typically line comments are used like this, which is acceptable for me:
```nix
{
  # Rust
  testRustSrc = ''
    ...
  '';
}
```

If your treesitter highlighting does not recognize this, you can ask upstream to support this hint position. It is possible because I [did it some years ago](https://github.com/oxalica/tree-sitter-nix/blob/master/queries/nvim-injections.scm)[^1] with nvim-treesitter queries.


[^1]: Though I'm not using it now because of the disappointment of their ecosystem, where every editors use a slightly different query language and editor is tightly coupled with a single version of parser implementation of each language.

oxalica commented on pkgs/development/tools/rust/rust-analyzer/test-neovim-lsp.nix:
```diff
@@ -1,62 +1,77 @@
+    # NB. Wait for server done type calculations before sending `hover` request,
+    # otherwise it would return `{unknown}`.
+    # Ref: https://github.com/rust-lang/rust-analyzer/blob/7b11fdeb681c12002861b9804a388efde81c9647/docs/dev/lsp-extensions.md#server-status
+    nvimConfig = # lua
```
Also this.

oxalica commented on pkgs/development/tools/rust/rust-analyzer/default.nix:
```diff
@@ -1,30 +1,41 @@
+  callPackage,
+  fetchFromGitHub,
+  rustPlatform,
+  CoreServices,
```
I believe this can be replaced by the new `darwin.apple_sdk_*` since #346043 (TL;DR: #354146). But I don't have a darwin machine. You can leave it as is if you cannot test it.

If this can be removed, we can also migrate `rust-analyzer*` to `pkgs/by-name`.

khaneliman replied:
I can do that, I didn't at first because I wasn't sure if you'd be concerned about the signature change during the end of the release cycle.

khaneliman committed:
rust-analyzer-unwrapped: 2024-11-04 -> 2024-11-11

khaneliman committed:
rust-analyzer-unwrapped: apple-sdk refactor

We don't need to define frameworks manually.

khaneliman committed:
rust-analyzer-unwrapped: move to by-name

khaneliman committed:
rust-analyzer: move to by-name

khaneliman commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 354304`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>ra-multiplex</li>
    <li>rust-analyzer</li>
    <li>rust-analyzer-unwrapped</li>
    <li>vscode-extensions.rust-lang.rust-analyzer</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>ra-multiplex</li>
    <li>rust-analyzer</li>
    <li>rust-analyzer-unwrapped</li>
    <li>vscode-extensions.rust-lang.rust-analyzer</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>ra-multiplex</li>
    <li>rust-analyzer</li>
    <li>rust-analyzer-unwrapped</li>
    <li>vscode-extensions.rust-lang.rust-analyzer</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>ra-multiplex</li>
    <li>rust-analyzer</li>
    <li>rust-analyzer-unwrapped</li>
    <li>vscode-extensions.rust-lang.rust-analyzer</li>
  </ul>
</details>

oxalica reviewed (approved):
Diff LGTM. Tested to work with neovim/coc.nvim.

<details><summary> `nixpkgs-review` result</summary>

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 354304`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>ra-multiplex</li>
    <li>rust-analyzer</li>
    <li>rust-analyzer-unwrapped</li>
    <li>vscode-extensions.rust-lang.rust-analyzer</li>
  </ul>
</details>

</details>
